File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.HashMap;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        HashMap<String, Integer> participantMap = new HashMap<String, Integer>();
        HashMap<String, Integer> completionMap = new HashMap<String, Integer>();
        
        for(String temp: participant){
            if(participantMap.containsKey(temp)){
                int tempInt = participantMap.get(temp);
                participantMap.put(temp, tempInt+1);
            }else{
                participantMap.put(temp, 1);
            }
        }        
        for(String temp: completion){
            if(completionMap.containsKey(temp)){
                int tempInt = completionMap.get(temp);
                completionMap.put(temp, tempInt+1);
            }else{
                completionMap.put(temp, 1);
            }
        }
        // System.out.println(participantMap);
        
        for(String temp: completion){
            // System.out.println(temp);
            if(participantMap.containsKey(temp)){
                int tempInt = participantMap.get(temp);
                if (tempInt>1){
                    participantMap.put(temp, tempInt-1);
                }else{                    
                    participantMap.remove(temp);
                }
            }
        }
        for(String temp: participantMap.keySet()){
            return temp;
        }
        return answer;
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    var answer = '';
    
    const map = new Map();
    
    for(let p of participant){
        if(!map.has(p)){
            map.set(p, 1)
        }else{
            map.set(p, map.get(p)+1)
        }
    }
    
    for(let c of completion){
        map.set(c, map.get(c)-1);
    }

    for(let k of map.keys()){
        if(map.get(k)==1) return k
    }
    
    return answer;
}